main/src/ca/uwaterloo/flix/language/ast/TypedAst.scala

```diff
@@ -214,7 +214,7 @@ object TypedAst {
 
     case class Do(op: Ast.OpSymUse, exps: List[Expr], tpe: Type, eff: Type, loc: SourceLocation) extends Expr
 
-    case class InvokeMethod2(exp: Expr, name: Name.Ident, exps: List[Expr], tpe: Type, eff: Type, loc: SourceLocation) extends Expr
+    case class InvokeMethod2(exp: Expr, name: Name.Ident, exps: List[Expr], methodTpe: Type, tpe: Type, eff: Type, loc: SourceLocation) extends Expr
```

Comment:
We can actually now remove InvokeMethod2 from TypedAst.

The idea is that we have InvokeMethod2 in all frontend ASTs up to TypedAst, but in TypeReconstruction we will translate InvokeMethod2 into InvokeMethod.

This will also mean some phases dont even have to worry about the new InvokeMethod2.

Reply:
So try to remove from here and see what happens.

Reply:
I removed it and a whole refactor is ongoing. Currently, invokeMethod2 is being removed from everything after Typer.
When does TypeReconstruction exactly happens?
If I understand correctly, it should happen right after/in TypedAst to translate invokeMethod2 to invokeMethod.
@magnus-madsen

Reply:
You can check the latest commit to see the changes, I am now working on the most important part in type reduction.

Reply:
I believe TypeReconstruction converts a KindedAst to a TypedAst. Is that not right?

Reply:
I see, thank you.